File: dotnet/Lexy/Lexy.Compiler/Language/Expressions/ArgumentTokenParseResult.cs
using System.Collections.Generic;
using Lexy.Compiler.Parser;

namespace Lexy.Compiler.Language.Expressions
{
    public class ArgumentTokenParseResult
    {
        public string ErrorMessage { get; }
        public bool IsSuccess { get; }
        public IEnumerable<TokenList> Result { get; }

        private ArgumentTokenParseResult(IEnumerable<TokenList> result)
        {
            Result = result;
            IsSuccess = true;
        }

        private ArgumentTokenParseResult(bool success, string errorMessage)
        {
            ErrorMessage = errorMessage;
            IsSuccess = success;
        }

        public static ArgumentTokenParseResult Success(IEnumerable<TokenList> result = null)
        {
            return new ArgumentTokenParseResult(result ?? new TokenList[]{});
        }

        public static ArgumentTokenParseResult Failed(string errorMessage)
        {
            return new ArgumentTokenParseResult(false, errorMessage);
        }
    }
}

File: typescipt/existing/src/language/expressions/argumentTokenParseResult.ts



namespace Lexy.Compiler.Language.Expressions;

public sealed class ArgumentTokenParseResult : ParseResult<IEnumerable<TokenList>>
{
   private ArgumentTokenParseResult(IEnumerable<TokenList> result) : base(result)
   {
   }

   private ArgumentTokenParseResult(bool success, string errorMessage) : base(success, errorMessage)
   {
   }

   public static ArgumentTokenParseResult Success(IEnumerable<TokenList> result = null)
   {
     return new ArgumentTokenParseResult(result ?? new TokenList[] { });
   }

   public static ArgumentTokenParseResult Failed(string errorMessage)
   {
     return new ArgumentTokenParseResult(false, errorMessage);
   }
}
